mailpoet/assets/js/src/settings/store/create_reducer.ts

```diff
@@ -16,13 +16,18 @@ export function createReducer(defaultValue: State) {
     action: Action,
   ): State => {
     switch (action.type) {
-      case 'SET_SETTING':
-        return setWith(
+      case 'SET_SETTING': {
+        const newState = setWith(
           clone(state),
           ['data', ...action.path],
           action.value,
           clone,
         );
+
+        newState.save.hasUnsavedChanges = true;
```

Comment:
Maybe instead of always setting true here, we could use `!_.isEqual(newState.data, state.originalData)` and avoid doing it in the selector method.